docs/extensions/guides/protocol-handlers.md

```diff
@@ -45,3 +45,9 @@ On the other hand, the subpath `"/display/notification"` would be routed to #3.
 
 The URI is routed to the most specific matching `pathSchema`.
 This way the `"/"` (root) `pathSchema` acts as a sort of catch all or default route if no other route matches.
+
+### Cleaning Up
+
+Currently there is not way to remove a protocol handler once it has been registered.
```

Comment:
`removeProtocolHandlers` removes *all* handlers. Yes calling `onProtocolRequest` would remove the handler. I didn't add a removal mechanism because I didn't think that was in the current design (though TBH I was never really given a design, I just asked some questions).

The alternative that I made for the old `removeProtocolHandlers` was to just call it directly for the extension developer.

Reply:
+1 from me! I think the ability to remove is something we need to provide. Can you add an event handler without ever being able to remove it...? If an extension wanted to stop listening to protocol requests, they'd have to implement some sort of mechanism to know to _ignore_ the requests. That's not very nice.

My vote would be for not overloading the meaning of `onProtocolRequest()` to also handle the removal. I think we should make removal explicit with a "remove" method.

Actually, it feels very much like adding/removing event handlers. The most intuitive thing there would be to rename `onProtocolRequest()` (which, frankly, sounds like a _handler_ itself, not the method you call to _add_ a handler) to `addProtocolHandler()`, and then provide a `removeProtocolHandler()` method.

Reply:
Sounds good I can do that.

Reply:
`onProtocolRequest` is more the name you'd give to a React prop on a component that provides a protocol event. But there, you assign your handler to it, and if you don't want to handle the event anymore, you assign `null/undefined` to it.

In this case, it's a method, so I think a different naming scheme would be more appropriate...

Here's what it looks like in my extension, for example:
https://github.com/Mirantis/lens-extension-cc/blob/master/src/renderer.tsx#L128-L131

It would feel more natural to me if it were:
```javascript
      this.addProtocolHandler(
        `/${EXT_EVENT_ACTIVATE_CLUSTER}`,
        this.handleProtocolActivateCluster
      );
      ...
      this.removeProtocolHandler(this.handleProtocolActivateCluster);
```

Oh, and I guess I'm advocating for a way to remove __individual__ handlers, not just all handlers (the hackweek branch just gave us `removeProtocolHandlers()` -- like all or nothing).

Reply:
I know you are asking for removing individual handlers. However, I won't support removing handler by the function because I am sure that it might come up that an extension wants to use the same handler for multiple routes.

Reply:
Then the `removeProtocolHandler()` can require the route as well.